### DISCUSSION TITLE
Testing script

### DISCUSSION THREAD

**Discussion by azzamsa** [nushell/nushell#6525] [Ideas]
Hi.

When making a PR, there are there command that repeated over and over. 

```
- [] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [] `cargo test --workspace --features=extra` to check that all the tests pass
```

Would it be nice if we put it into into one script such `test.nu` à la `build-all.nu`?
If so. I would make a PR.

**Comment by fdncred** [comment]
Here are my thoughts.

1. Some of these may be problematic to be in a nushell script because that infers that nushell will run it. On Windows, if you compile a program that's already running in a way that it overwrites the running program, you get sharing violations and it won't compile.
2. I have `cargo fmt --all` setup in vscode so that it happens on save. So, really, I never run this command at all.
3. I rely on clippy and test (i actually use nextest) to be in my history.

Having said that, I don't really have objections with a nushell script containing all of this in the root of the repo, if it helps people. At a minimum it could serve as documentation to look at when you don't remember the syntax of how to run these commands. Also, at some point we may want to create a build folder to organize all these build, install, dev, register scripts better though.
